lib/kube/proxy/resource_filters_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"path/filepath"
+	"sort"
+	"testing"
+	"text/template"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	authv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/kube/proxy/responsewriters"
+)
+
+func Test_filterBuffer(t *testing.T) {
+	log := logrus.New()
+	type objectAndAPI struct {
+		obj string
+		api string
+	}
+	teleToKubeResource := map[string]objectAndAPI{
+		types.KindKubePod:                   {obj: "Pod", api: "v1"},
+		types.KindKubeSecret:                {obj: "Secret", api: "v1"},
+		types.KindKubeConfigmap:             {obj: "ConfigMap", api: "v1"},
+		types.KindKubeService:               {obj: "Service", api: "v1"},
+		types.KindKubeServiceAccount:        {obj: "ServiceAccount", api: "v1"},
+		types.KindKubePersistentVolumeClaim: {obj: "PersistentVolumeClaim", api: "v1"},
+		types.KindKubeDeployment:            {obj: "Deployment", api: "apps/v1"},
+		types.KindKubeReplicaSet:            {obj: "ReplicaSet", api: "apps/v1"},
+		types.KindKubeStatefulset:           {obj: "StatefulSet", api: "apps/v1"},
+		types.KindKubeDaemonSet:             {obj: "DaemonSet", api: "apps/v1"},
+		types.KindKubeRole:                  {obj: "Role", api: "rbac.authorization.k8s.io/v1"},
+		types.KindKubeRoleBinding:           {obj: "RoleBinding", api: "rbac.authorization.k8s.io/v1"},
+		types.KindKubeCronjob:               {obj: "CronJob", api: "batch/v1"},
+		types.KindKubeJob:                   {obj: "Job", api: "batch/v1"},
+		types.KindKubeIngress:               {obj: "Ingress", api: "networking.k8s.io/v1"},
+	}
+
+	_, decoder, err := newEncoderAndDecoderForContentType(responsewriters.DefaultContentType, newClientNegotiator())
+	require.NoError(t, err)
+
+	type args struct {
+		contentEncoding string
+		dataFile        string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "Test filterBuffer with gzip",
```

Comment:
The name of this test as Go sees it and as it will appear in the logs will be `Test_filterBuffer/Test_filterBuffer_with_gzip`. You probably don't need to say `Test_filterBuffer` twice.

```suggestion
			name: "with gzip",
```